### PR TITLE
undo unintended v1 -> v2 hpopt search space changes

### PR DESCRIPTION
**Pull request by KnathanM**
Addresses #1207 with 154172d

Note that the depth parameter in this PR has uniform probability (i.e. 2 is as likely to be chosen as 3), which is different than v1, see the linked issue. In this PR the ffn_hidden_dim parameter still puts less probability on 300 than 400, which matches v1. I would have changed it but don't see a good way to do this. `tune.choice` prevents the tuner from knowing that `400>300` I think. We could chose a number between 3 and 24+1 and then multiply by 100, but I don't know how to do that with `tune`. 

I also left ffn_num_layers alone. In v1 you could get a model with 0 hidden layers and no non-linear activations (just a single nn.Linear), but in v2 the smallest is Linear -> dropout -> non-linear -> Linear, which feels okay to me. 

In bf5b821, I group the parameters by which submodule they belong to (all, mp, agg, ffn, optimizer). 

In 81a6a69, I add back *.ipynb to the .gitignore because otherwise my `black` will not ignore them when I format the whole chemprop dir `black .`. It was removed in #1226 because we didn't know why it was there.

### CODE REVIEW

**Comment by craabreu** [comment]
@KnathanM, can you please help me understand a few points?

1. Is the following comment from [Ray's doc](https://docs.ray.io/en/latest/tune/api/search_space.html) accurate?

```py
    # Sample a random uniformly between -21 (inclusive) and 12 (inclusive (!))
    # rounding to multiples of 3 (includes 12)
    # if q is 1, then randint is called instead with the upper bound exclusive
    "qrandint": tune.qrandint(-21, 12, 3),
```

2. What is the drawback of sampling `ffn_hidden_dim` from a categorical distribution? Or, conversely, how much does the tuner benefit from knowing that 400>300?

**Comment by JacksonBurns** [comment]
@craabreu for 2. it depends what tuner one uses. Some may have a sense of space between 400 and 300, others will not.

**Review by JacksonBurns — APPROVED** [review]
I think this is good to go. The opinionated reordering looks good me, and the ranges of hyperparameters being allowed are reasonable (and of course, what we had before, which is nice). I think further changes to this will be 'opinionated' to the level that, if one wishes to optimize to some other bounds, they should just adapt this code to do so.

**Review by akshatzalte — APPROVED** [review]
LGTM!
